### ISSUE TITLE
Develop a fixed sample of using Mitiq consuming an IR

### ISSUE THREAD

**Issue by crazy4pi314** [unitaryfund/mitiq#1208]
Pre-Request Checklist
---------------------

- [ ] I checked to make sure that this feature has not already been requested.

Issue Description
-----------------

Give a detailed description of the new feature, and what problem it is trying to solve.

If this is a proposal to change an existing feature, also outline the current
implementation and its drawbacks.

Thanks for helping us improve Mitiq! 🙂

Proposed Solution
-----------------

Let us know what you'd like to see happen, and be sure to mention
any alternatives that you've considered.

Additional References
---------------------

If applicable, provide some references that will help us better understand the request.

**Comment by ACE07-Sev** [comment]
@natestemen I think closest to this would just be `quick`. I have this IR mentioned here as `quick.circuit.Circuit`, and other frameworks are implemented as sub-classes. I could build sth similar for mitiq as well if you'd prefer sth smaller.

**Comment by natestemen** [comment]
By QIR we literally mean [QIR](https://github.com/qir-alliance/) here rather than "some quantum intermediate representation".

**Comment by natestemen** [comment]
> Have you considered using qrisp by the way?

Not really, but only because I'm not very familiar with it. The README does say "Qrisp is a high-level quantum programming framework" which makes me a bit confused as to how this is related to IRs.

We don't have much standardization across the industry, but AFAIK QIR is the IR with the most institutional backing, so I think it's the best place to start.
